dask/dataframe/core.py

```diff
@@ -5964,7 +5964,7 @@ def repartition_npartitions(df, npartitions):
             if isinstance(divisions, np.ndarray):
                 divisions = divisions.tolist()
 
-            divisions = list(divisions)
+            divisions = sorted(set(divisions))
```

Comment:
Note that the last two values can reasonably be the same

Reply:
I'm not sure if in the branch starting at line 5941 we could end up with duplicate last two values based on how the interpolation is done at line 5952? The next two lines set the first and last divisions to be the first and last values in `df.divisions`